mxcubecore/HardwareObjects/GenericDiffractometer.py

```diff
@@ -1091,14 +1096,43 @@ def move_to_beam(self, x, y, omega=None):
                 "Diffractometer: could not center to beam, aborting"
             )
 
-    def image_clicked(self, x, y, xi=None, yi=None):
-        """
-        Descript. :
-        """
-        if self.use_sample_centring:
-            sample_centring.user_click(x, y)
-        else:
-            self.user_clicked_event.set((x, y))
+    def image_clicked(self, x: float, y: float, xi=None, yi=None):
+        """Handles a user click sent from the frontend during the manual centring.
+
+        This method is called by the backend when the user clicks on the sample
+        image in the frontend.
+
+        The attribute `self.waiting_for_click` controls whether the click should
+        be accepted or ignored:
+          - None: click is accepted (legacy)
+          - True: waiting for a click, accept it and mark it as received
+          - False: already received a click, ignore further clicks
+
+        Args:
+            x (float): X coordinate of the click.
+            y (float): Y coordinate of the click.
+            xi (float, optional): ...
+            yi (float, optional): ...
```

Comment:
Types should preferably be documented as type hint annotations, rather than docstrings.

```suggestion
    def image_clicked(self, x: float, y: float, xi: float | None=None, yi: float | None=None):
        """Handles a user click sent from the frontend during the manual centring.

        This method is called by the backend when the user clicks on the sample
        image in the frontend.

        The attribute `self.waiting_for_click` controls whether the click should
        be accepted or ignored:
          - None: click is accepted (legacy)
          - True: waiting for a click, accept it and mark it as received
          - False: already received a click, ignore further clicks

        Args:
            x: X coordinate of the click.
            y: Y coordinate of the click.
            xi: ...
            yi: ...
```